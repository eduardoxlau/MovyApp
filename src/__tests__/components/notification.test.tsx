/* eslint-disable no-undef */
import { render } from '@testing-library/react';

import Notification from 'components/notification';

describe('<Notification />', () => {
  it('should render snapshot Notification', () => {
    const { asFragment } = render(<Notification message="" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should appear Notification', () => {
    render(<Notification message="text" />);
    expect(document.querySelector('h1')).toBeVisible();
  });
  it('shouldn"t appear Notification', () => {
    jest.useFakeTimers();
    render(<Notification message="" />);
    jest.runAllTimers();
    expect(document.querySelector('.hidden')).toBeInTheDocument();
  });
});
