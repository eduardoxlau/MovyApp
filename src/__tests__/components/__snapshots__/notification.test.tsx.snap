// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> should render snapshot Notification 1`] = `
<DocumentFragment>
  <h1
    class="w-full p-2 rounded bg-red-400 flex"
  />
</DocumentFragment>
`;
